chalk-ir/src/lib.rs

```diff
@@ -2198,14 +2458,30 @@ pub enum GoalData<I: Interner> {
 }
 
 impl<I: Interner> GoalData<I> {
+    /// Create an interned goal.
     pub fn intern(self, interner: &I) -> Goal<I> {
         Goal::new(interner, self)
     }
 }
 
+/// Kinds of quantifiers in the logic, such as `forall` and `exists`.
 #[derive(Copy, Clone, Debug, PartialEq, Eq, Hash, PartialOrd, Ord)]
 pub enum QuantifierKind {
+    /// Universal quantifier `ForAll`.
+    ///
+    /// A formula with the universal quantifier `forall(x). P(x)` is satisfiable
+    /// if and only if the subformula `P(x)` is true for all possible values for x.
     ForAll,
+
+    /// Existential quantifier `Exists`.
+    ///
+    /// A formula with the existential quantifier `exists(x). P(x)` is satisfiable
+    /// if and only if there exists at least one value for all possible values of x
+    /// which satisfies the subformula `P(x)`.
+
+    /// In the context of chalk, the existential quantifier usually demands the
+    /// existence of exactly one instance (i.e. type) that satisfies the formula
+    /// (i.e. type constraints). More than one instance means that the result is ambiguous.
```

Comment:
Mmm, this is...somewhat correct. If using the slg solver with `solve_multiple`, you'll get back multiple answers. But this is right in that it *does* say "usually"

Reply:
Might not have been the best idea to copy this part from the book then (though it's sometimes tricky to decide what should go where)

Reply:
I think this is fine, the "usually" makes it so.